### ISSUE TITLE
Support the use of AWS EKS IAM roles for service accounts (IRSA)

### ISSUE THREAD

**Issue by lambda-9** [elastic/helm-charts#681]
**Describe the feature:**
Support the use of AWS EKS IAM roles for service accounts.

This requires two things to take place:

1. Filebeat (and potentially other charts) do not appear to be use an AWS SDK version that supports assuming an IAM role via an OIDC web identity token file ([sts:AssumeRoleWithWebIdentity](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html)). This is required in order to use EKS's [IAM roles for service accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) feature. AWS SDK version requirements are [here](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts-minimum-sdk.html). 

2. Charts also need to support service account annotations described in issue #627 

**Describe a specific use case for the feature:**
Provide AWS IAM roles to pods instead of supplying credentials directly in configuration. Adding the following annotation to the service account would allow the use of the corresponding role.

```
annotations:
  eks.amazonaws.com/role-arn: arn:aws:iam::<ACCOUNT-ID>:role/<ROLE-NAME>
```

**Comment by jim-barber-he** [comment]
I'd like to add that the permissions check on the mounted token in the entrypoint script for ElasticSearch also needs to change.

It currently enforces a permission of 400 or 600 and will exit if that is not the case.
But the ElasticSearch containers do not run as the root user, so that permission on the mounted token means the IAM role details cannot be read.
For IAM roles for service accounts to work, the pod's `securityContext.fsGroup` must be set to the GID of the user running in the containers, which results in the mounted token being owned by the GID and given permission 640 (group read).
If the entrypoint allowed 640, then I suspect things may work.
I mentioned this in the following issue:
https://github.com/elastic/elasticsearch/issues/52625#issuecomment-632587920

**Comment by jmlrt** [comment]
@lambda-9 Thanks for submitting this request.

- Service account annotations have been implemented in #686 
- Requests related to AWS SDK integration should be adressed to the products repositories (example elastic/elasticsearch#52625 for Elasticsearch)

Is there anything else required at Helm charts level or can we close this issue? (cc @jim-barber-he)

**Comment by lambda-9** [comment]
No, this can be closed. Thank you!
